src/Miunie.Discord/CommandModules/ProfileCommand.cs

```diff
@@ -30,19 +30,26 @@ public async Task ShowProfileAsync(CommandContext ctx, MiunieUser m = null)
         }
 
         [Command("+rep")]
-        public async Task AddReputationAsync(CommandContext ctx, MiunieUser m)
+        public async Task AddReputationAsync(CommandContext ctx, MiunieUser m, [RemainingText] string reason = "")
         {
             var source = _entityConvertor.ConvertUser(ctx.Member);
             var channel = _entityConvertor.ConvertChannel(ctx.Channel);
-            await _profileService.GiveReputationAsync(source, m, channel);
+            await _profileService.GiveReputationAsync(source, m, channel, reason);
         }
 
         [Command("-rep")]
-        public async Task RemoveReputationAsync(CommandContext ctx, MiunieUser m)
+        public async Task RemoveReputationAsync(CommandContext ctx, MiunieUser m, [RemainingText] string reason = "")
         {
             var source = _entityConvertor.ConvertUser(ctx.Member);
             var channel = _entityConvertor.ConvertChannel(ctx.Channel);
-            await _profileService.RemoveReputationAsync(source, m, channel);
+            await _profileService.RemoveReputationAsync(source, m, channel, reason);
+        }
+
+        [Command("repHistory")]
```

Comment:
`[Command("repHistory"), Alias("repLog")]`
Maybe introducing this alias here wouldn't be bad.